network/src/connectivity_manager/mod.rs

```diff
@@ -473,6 +491,54 @@ fn log_dial_result(peer_id: PeerId, addr: NetworkAddress, dial_result: DialResul
     }
 }
 
+impl PeerAddresses {
```

Comment:
it seems the only need for this struct is to impl Display? I think it's proper to impl Deref for HashMap for this

Reply:
what about `as_inner` / `as_inner_mut`?  i think `Deref` trait is kinda hard to read and seems to break rust-analyzer autocomplete...

Reply:
hmm, actually I'll just use `.0` directly